### DISCUSSION TITLE
batch inference

### DISCUSSION THREAD

**Discussion by liuxiaohao-xn** [abetlen/llama-cpp-python#347] [Q&A]
I have multiple prompts, I want to feed them all at once to the model to generate the output，can you tell me how to achieve it ?

**Comment by ilyalasy** [comment]
Same question here, it seems llama.cpp already supports batched inference.
See [baby-llama](https://github.com/ggerganov/llama.cpp/blob/master/examples/baby-llama/baby-llama.cpp#L759-L770).
See [this](https://github.com/ggerganov/llama.cpp/discussions/1220#:~:text=Add%20support%20for%20%22batch%20inference%22) discussion.

@abetlen It seems this usecase is very popular, what do you think?

**Reply by abetlen** [reply]
Currently the llama.h api does not support efficient batched inference. The babyllama example with batched inference uses the ggml api directly which this binding does not (I am working on a seperate project that does that but ggml repo is slightly outdated between ggml <-> llama.cpp). I'll take a look at some options for how to implement this though.

**Reply by ilyalasy** [reply]
Alright, thanks!

**Comment by origintopleft** [comment]
What kind of work would need to be done, either here or upstream in llama.cpp, to get batch inference fully working? Is there a roadmap anywhere already?

I've got some code here from work that needs my help working against Vicuna, and the only real obstacle left is getting batch processing working

**Reply by origintopleft** [reply]
I should probably clarify: I was asking because I was hoping to possibly contribute some help to the project :sweat_smile:
